index.js

```diff
@@ -1,62 +1,109 @@
 'use strict';
 
+// Built-in types
 var types = [
   'object',
   'number',
   'string',
   'symbol',
   'boolean',
+  'date',
   'function', // Weird to expose this
-  'undefined',
+  'undefined', // And this?
 ];
 
-function normalize(type, value) {
-  var args = Array.prototype.slice.call(arguments, 2);
 
-  var isType = conforms(type, value);
-  var isFunction = typeof value === 'function';
+function normalize(coercer, value) {
+  if (typeof value === 'function') {
+    if (coercer === 'function') {
+      return value;
+    }
+    value = value.apply(null, Array.prototype.slice.call(arguments, 2));
+  }
+  return coerce(coercer, value);
+}
+
+
+function coerce(coercer, value) {
 
-  if (!isType && !isFunction) {
-    return null;
+  // Handle built-in types
+  if (typeof coercer === 'string') {
+    if (coerce[coercer]) {
+      return coerce[coercer](value);
+    }
+    return fallback(coercer, value);
   }
 
-  if (isType) {
-    return value;
+  // Handle custom coercer
+  if (typeof coercer === 'function') {
+    return coercer(value);
   }
 
-  var result = value.apply(null, args);
+  // Array of coercers, try in order until one returns a non-null value
+  var result = null;
+  coercer.some(function(coercer) {
+    result = coerce(coercer, value);
+    return result !== null;
+  });
+
+  return result;
+}
+
+
+coerce.string = function(value) {
+  value = primitive(value);
+  if (value != null &&
```

Comment:
But won't the `primitive` call above it turn the object into whatever `valueOf` returns?  Wouldn't that have to return another object with the `toString` method?